library/std/src/sys/windows/process.rs

```diff
@@ -166,6 +166,7 @@ pub struct Command {
     stdout: Option<Stdio>,
     stderr: Option<Stdio>,
     force_quotes_enabled: bool,
+    proc_thread_attributes: BTreeMap<usize, ProcThreadAttributeValue>,
```

Comment:
I've skimmed over this and previous pr and don't understand why this can't be simply HashMap, is some order required here?

Reply:
In case of `HashMap`, this field should be `Option`, to skip creating  `HashMap`, if it will be unused.

Reply:
The original author had it in there, but to be honest I also don't get why this isn't a HashMap.

Reply:
Hey, thanks for continuing this work.

I originally used a BTreeMap here because I understood this to be better than a HashMap for small collections with simple integer keys. I don't know that it matters a ton, but if using a HashMap, it's probably best to use a hash like FxHash. I don't think hash collisions are a concern here (no one should be passing in untrusted keys) so minimizing the hashing time would be ideal.

Reply:
> Hey, thanks for continuing this work.

No problem, I actually had the same plan as you to add this feature to the standard lib, but then @ChrisDenton pointed out to me that you already started this, so thanks to you.

Reply:
> I don't know that it matters a ton, but if using a HashMap, it's probably best to use a hash like FxHash.

Is that available in std lib?

Reply:
> Is that available in std lib?

Oh, I guess not. Wasn't even thinking about that.

At the end of the day, I don't actually *know* what would be faster here, but it seemed worth it to me to avoid hashing altogether for this use-case.

Reply:
Kind of true, but then we would have to have some sort of `Vec<(usize, ProcThreadAttributeValue)>` structure and check every time if the attribute is already inside the vec.

Reply:
Right, hence my original thought to use a BTreeMap instead for a map API without hashing.

But again, I don't actually know which is faster, and I doubt it makes a noticeable difference anyway.

Reply:
https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwip2bzp1emAAxXStKQKHTQWCz0QFnoECBcQAQ&url=https%3A%2F%2Fusers.rust-lang.org%2Ft%2Fhashmap-vs-btreemap%2F13804&usg=AOvVaw1KdHwoi4y3SmgX2Bq_et5V&opi=89978449

https://github.com/maciejhirsz/ramhorns/issues/62#issue-1382015785

Reply:
`BTreeMap` seams to be faster overall, but i don't know if it holds on a small scale.

Reply:
On balance, I think using `BTreeMap` is the best call.

I don't think performance is the main concern as there are likely to be relatively few attributes set and spawning a process isn't exactly fast anyway (though it is a minor concern). However, my main reason for preferring `BTreeMap` is that it's in `alloc`. Ideally `library/src/sys/*` should depend only on `core` and `alloc`. Then dependencies are nice and linear: core -> alloc -> sys -> std. We currently violate this in a number of places but simplifying the nest of inter-dependencies is a goal.